kvdb-rocksdb/src/lib.rs

```diff
@@ -189,58 +189,37 @@ impl Default for DatabaseConfig {
 	}
 }
 
-/// Database iterator (for flushed data only)
-// The compromise of holding only a virtual borrow vs. holding a lock on the
-// inner DB (to prevent closing via restoration) may be re-evaluated in the future.
-//
-pub struct DatabaseIterator<'a> {
-	iter: InterleaveOrdered<::std::vec::IntoIter<(Box<[u8]>, Box<[u8]>)>, DBIterator>,
-	_marker: PhantomData<&'a Database>,
+struct DBAndColumns {
+	db: DB,
+	column_names: Vec<String>,
```

Comment:
we have to store column names instead of columns, because [`cf_handle`](https://docs.rs/rocksdb/0.13.0/rocksdb/struct.DB.html#method.cf_handle) returns a reference to ColumnFamily.